dashboard/app/models/course.rb

```diff
@@ -212,6 +237,34 @@ def summarize
     }
   end
 
+  # If a user has no experiments enabled, return the default set of scripts.
+  # If a user has an experiment enabled corresponding to an alternate script in
+  # this course, use the alternate script in place of the default script with
+  # the same position.
+  # @param user [User]
+  # @return [Array<Script>]
+  def scripts_for_user(user)
+    return default_scripts unless user
+    default_course_scripts.map do |cs|
+      select_course_script(user, cs).script
+    end
+  end
+
+  # Return the first alternate course script with a default script matching
+  # default_course_script, and for which the user has the corresponding
+  # experiment enabled, if one exists. Otherwise return the default course
+  # script.
+  # @param user [User]
+  # @param default_course_script [CourseScript]
+  # @return [CourseScript]
+  def select_course_script(user, default_course_script)
+    alternates = alternate_course_scripts.where(default_script: default_course_script.script).all
+    alternates.each do |cs|
+      return cs if SingleUserExperiment.enabled?(user: user, experiment_name: cs.experiment_name)
+    end
+    return default_course_script
```

Comment:
I wonder if this would be better expressed as
``` ruby
matching_non_default = alternatives.find{|cs| SingleUserExperiment.enabled?(user: user, experiment_name: cs.experiment_name)}
matching_non_default || default_course_script
```

This might mostly be a side effect of me seeing lots of forEach type loops in JS, but in my first I was thinking that the return on line 263 only returned out of that block (rather than for the entire function)

Reply:
`ifnone` seems to be a standard part of a standard libary, so I'm going with it.

Reply:
Looks like ifnone needs to be a function, which is less convenient. so never mind.